docs/externalnode.md

```diff
@@ -0,0 +1,309 @@
+The configuration for antrea-controller is modified in the `antrea-config` ConfigMap
+like the following options for the feature to work:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: antrea-config-dcfb6k2hkm
+  namespace: kube-system
+data:
+  antrea-controller.conf: |
+    featureGates:
+      ExternalNode: true
+```
+
+Since antrea-agent is running on an external Node which is not managed by Kubernetes,
+a single configuration file is required to locate on each machine where antrea-agent
+is running. The following options for the feature are needed:
+```yaml
+featureGates:
+  ExternalNode: true
+nodeType: externalNode
+```
+Other options for antrea-agent running on an external Node please refer to the sample
+configuration [antrea-agent.conf](../build/yamls/externalnode/conf/antrea-agent.conf).
+
+## The ExternalNode resource
+
+A typical ExternalNode resource example:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ExternalNode
+metadata:
+  name: vm1
+  namespace: vm-ns
+  labels:
+    role: db
+spec:
+  interfaces:
+    - ips: ["172.16.100.3"]
+      name: ""
+```
+
+**Note**: `ExternalNode` resource is working in `namespace` scope.
+
+### Interfaces
+
+The `interfaces` field specifies the list of the network interfaces expected to be
+guarded by Antrea NetworkPolicy. One interface at least is required. The `name` or
+`ips` can be used to identify the target interface. **The field `ips` must be
+provided in the CRD**, but `name` is optional. Multiple IPs on a single interface is
+supported.
+
+A single `ExtrenalEntity` is created mapping to one interface, and antrea-controller
+performs the convention when it finds an `ExternalNode` is created.
+The created `ExternalEntity` has the following characteristics:
+
+- It is configured within the same namespace as the `ExternalNode`.
+- The `name` is generated according to the following principles:
+    - Use the `ExternalNode` name directly, if there is only one interface, and its
+      name is not specified.
+    - Use this format `$ExternalNode.name-$hash($interface.name)[:5]` for other cases.
+- The `externalNode` field is set with the `ExternalNode` name.
+- The `owner` is referring to the `ExternalNode` object.
+- All labels added on `ExternalNode` are copied to the `ExternalEntity`.
+- One Endpoint is created for per IP in the interface, and every Endpoint's name uses
+  the interface name if it is set.
+
+A sample of the `ExtrenalEntity` created for the above `ExternalNode` interface is,
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: ExternalEntity
+metadata:
+  labels:
+    role: db
+  name: vm1
+  namespace: vm-ns
+  ownerReferences:
+  - apiVersion: v1alpha1
+    kind: ExternalNode
+    name: vm1
+    uid: 99b09671-72da-4c64-be93-17185e9781a5
+  resourceVersion: "5513"
+  uid: 5f360f32-7806-4d2d-9f36-80ce7db8de10
+spec:
+  endpoints:
+  - ip: 172.16.100.3
+  externalNode: vm1
+```
+
+Note: **Only one interface is supported since Antrea v1.8**.
```

Comment:
**Only one interface is supported for Antrea v1.8**.

Reply:
done.